tsl/src/continuous_aggs/create.c

```diff
@@ -2409,8 +2388,11 @@ cagg_rebuild_view_definition(ContinuousAgg *agg, Hypertable *mat_ht)
 
 	mattablecolumninfo_init(&mattblinfo, NIL, NIL, copyObject(direct_query->groupClause));
 	finalizequery_init(&fqi, direct_query, &mattblinfo);
-	RangeTblEntry *usertbl_rte = list_nth(direct_query->rtable, 0);
-	mattablecolumninfo_addinternal(&mattblinfo, usertbl_rte, 0);
+	if (!agg->data.finalized)
```

Comment:
We don't need the `if (!agg->data.finalized)` here because it already `return`ed above with `if (agg->data.finalized)`.

Reply:
Fixed!